pgmpy/readwrite/BIF.py

```diff
@@ -0,0 +1,312 @@
+import numpy as np
+from pyparsing import Word, alphanums, Suppress, Optional, CharsNotIn, Group, nums, ZeroOrMore, OneOrMore, cppStyleComment, Literal
+from pgmpy.models import BayesianModel
+from pgmpy.factors import TabularCPD
+import re
+
+
+class BifReader(object):
+
+    """
+    Base class for reading network file in bif format
+    """
+    def __init__(self, path=None, string=None):
+
+        """
+        Initialisation of BifReader object
+
+        Parameters
+        ----------------
+        path : file or str
+                File of bif data
+        string : str
+                String of bif data
+        Examples
+        -----------------
+        # dog-problem.bif file is present at
+        # http://www.cs.cmu.edu/~javabayes/Examples/DogProblem/dog-problem.bif
+        >>> reader = BifReader("bif_test.bif")
+        """
+        multi_space = OneOrMore(Literal(' ')|'\t'|'\r'|'\f')                                    # An expression for matching multi spaces
+
+        if path:
+            self.network = open(path, 'r').read()                                               
+
+        elif string:
+            self.network = string        
+
+        else:
+            raise ValueError("Must specify either path or string")
+
+        if '"' in self.network:
+            """
+            Replacing quotes by spaces to remove case sensitivity like:
+            "Dog-Problem" and Dog-problem
+            or "true""false" and "true" "false" and true false
+            """
+            self.network = self.network.replace('"', ' ')
+
+        self.network = multi_space.setParseAction(lambda t:' ').transformString(self.network)   # replacing mulitple spaces or tabs by one space
+
+        if '/*' or '//' in self.network:
+            self.network = cppStyleComment.suppress().transformString(self.network)             # removing comments from the file
+
+        self.get_network_name()
+        self.get_variables_info()
+        self.get_cpd()
+        self.get_edges()
+        self.model = self.get_model()
+
+    def get_network_name(self):
+
+        """
+        Retruns the name of the network
+
+        network attributes are of the format
+            network <name_of_network> {
+                attirbutes
+            }
+        Example of network attribute
+        ------------------------------
+        network "Dog-Problem" { //5 variables and 5 probability distributions
+                property "credal-set constant-density-bounded 1.1" ;
+        }
+
+        Sample run
+        ---------------
+        >>> reader = BIF.BifReader("bif_test.bif")
+        >>> reader.network_name()
+        'Dog-Problem'
+        """
+        network_attribute = Suppress('network') + Word(alphanums+'_'+'-') + '{'                            # Creating a network attribute 
+        temp = network_attribute.searchString(self.network)
+        self.network_name = temp[0][0]
```

Comment:
@khalibartan Try to avoid defining properties in methods other than `__init__`. It helps in keeping things clear as you will be able to just see the `__init__` method and can tell what are the properties.

So here rather than assigning value to `self.network_name` just simply return `temp[0][0]` and in the `__init__` method rather that simply calling `self.get_network_name()` do something like `self.network_name = self.get_network_name()`. 
The end result is the same in both cases but the second method will make the code much more cleaner.